datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -425,29 +428,41 @@ def _store_labels(self, metric, scraper_config):
                         matching_value = label_value
                     elif label_name in scraper_config['label_joins'][metric.name]['labels_to_get']:
                         label_dict[label_name] = label_value
-                try:
-                    if scraper_config['_label_mapping'][matching_label].get(matching_value):
-                        scraper_config['_label_mapping'][matching_label][matching_value].update(label_dict)
+
+                if matching_value is not None:
+                    if scraper_config['use_namespaced_tag_matching']:
+                        # If we have a namespaced metric, we will only match inside same namespace
+                        namespace = sample[self.SAMPLE_LABELS].get(self.METRICS_NAMESPACE_LABEL, '')
+                        mapping_label = namespace + matching_label
```

Comment:
It's just a key, It would be pretty unlikely to have something without ns matching the name of something with ns, but I can add a separator just in case.